Common/UnitDefinitions/MassConcentration.json

```diff
@@ -12,8 +12,8 @@
       "SingularName": "GramPerCubicMillimeter",
       "PluralName": "GramsPerCubicMillimeter",
       "BaseUnits": {
-        "L": "Millimeter",
-        "M": "Gram"
+        "L": "Meter",
+        "M": "Kilogram"
```

Comment:
As mentioned in discussion, `Gram` and `Millimeter` is correct here. The point is that when doing `new MassConcentration(1, UnitSystem.SI)` it will look for any units with base units `Meter` and `Kilogram` (`KilogramPerCubicmeter`) and if found construct the quantity with that unit.

If you provide a custom `UnitSystem` with say `Length=Millimeter` and `Mass=Gram` then the above example would choose `GramPerCubicMillimeter` unit.

Reply:
Ok, this makes sense now... I hadn't considered custom UnitSystems before- but one could imagine having one per App, User, Screen or even one for persistence.. The only problem that comes to mind is the possible overlap of units, and the accidental change of behavior when extending existing units. It would be nice to be able to register the default units for (some) types inside a unit system- maybe even have it serve as cache for the As(UnitSystem unitSystem) method.

Reply:
`UnitSystem` is not my brainchild, @tmilnthorp can probably answer a lot better than me, but I believe the idea is to make it easier to construct quantities and perform arithmetic with consistent units of measurement - so an application can switch from say SI unit system to British Engineering unit system and all the GUI now renders with feet and pounds and calculations are performed with said system. In particular in applications where you work with pure numbers and not wrapped in quantity types like we have in Units.NET, I understand that choosing and sticking to a particular system is crucial to get consistent results.

The overlap of units is definitely something we need to figure out. I don't have the answer to how we should best solve that, but your idea of making it configurable sounds reasonable. As previously mentioned, this whole unit system thing is still kind of experimental so any feedback is very useful.

Reply:
It should probably be noted that if no base (default) unit is defined for a given type then the constructor for the quantity that takes a UnitSystem will fail with "No units were found" exception:

            Assert.Equal(new Length(1, UnitSystem.SI).Unit, Length.BaseUnit);  // BaseUnit = Meter 
            Assert.Equal(new Volume(1, UnitSystem.SI).Unit, Volume.BaseUnit);  // fails because no unit is defined
Result of the second test is System.ArgumentException : No units were found for the given UnitSystem.
Parameter name: unitSystem

PS We do expect that BaseUnit is always SI

Reply:
> Result of the second test is System.ArgumentException : No units were found for the given 

I believe that is simply because we haven't gotten around to adding `BaseUnits` to all our units yet. In https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Volume.json I see that only two units have it defined.